### PR TITLE
Test: Change randomValueOtherThan(null, supplier)

### LINKED ISSUE #27821: Test: randomValueOtherThan confusing with nulls
`ESTestCase#randomValueOtherThan` ignores nulls. This is it now:

```
    /**
     * helper to get a random value in a certain range that's different from the input
     */
    public static <T> T randomValueOtherThan(T input, Supplier<T> randomSupplier) {
        if (input != null) {
            return randomValueOtherThanMany(input::equals, randomSupplier);
        }

        return(randomSupplier.get());
    }
```

So if you pass in `null` it'll just run the supplier one time. I think this is confusing - if you pass in null it shouldn't return null. I think something like:

```
    /**
     * helper to get a random value in a certain range that's different from the input
     */
    public static <T> T randomValueOtherThan(T input, Supplier<T> randomSupplier) {
        return randomValueOtherThanMany(v -> Object.equals(input, v), randomSupplier);
    }
```

Would be better. Discuss?

### PR DESCRIPTION
**Pull request by nik9000**
When the first parameter of `ESTestCase#randomValueOtherThan` is `null`
then run the supplier until it returns non-`null`. Previously,
`randomValueOtherThan` just ran the supplier one time which was
confusing.

Unexpectedly, it looks like not tests rely on the original `null`
handling.

Closes #27821

### CODE REVIEW

File: test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java [hunk 1]
```diff
@@ -166,4 +167,18 @@ public void testRandomUniqueTotallyUnique() {
...
+        int bad = randomInt();
+        assertNotEquals(bad, (int) randomValueOtherThan(bad, ESTestCase::randomInt));
+
+        /*
+         * "funny" way of calling where the value is null. This once
+         * had a unique behavior but at this point `null` acts just
+         * like any other value.
+         */
+        Supplier<Object> usuallyNull = () -> usually() ? null : randomInt();
+        assertNotNull(null, randomValueOtherThan(null, usuallyNull));
```
Comment (jasontedor) [inline test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java]: Why pass a null message here, I think `assertNotNull(randomValueOtherThan(null, usuallyNull));` is fine here, the expectation will fail saying we got `null`?

**Review by jasontedor — APPROVED** [review]
I left one comment but LGTM otherwise.

File: test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java (hunk 1 above)
Comment (nik9000) [reply test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java]: My mistake. I originally had `assertThat` and wasn't super careful reworking it. Strange that a null message is allowed though....

**Comment by nik9000** [comment]
Thanks for reviewing @jasontedor! I'll merge now and backport soon!

**Pull request merged**